src/test/java/org/owasp/esapi/logging/appender/LogPrefixAppenderTest.java

```diff
@@ -159,8 +186,26 @@ private void runTest(String typeResult, String userResult, String clientResult,
 
         //Since everything is mocked these booleans don't much matter aside from the later verifies
         LogPrefixAppender lpa = new LogPrefixAppender(false, false, false, false, null);
-        String result =   lpa.appendTo(testLoggerName, testEventType, testLogMessage);
 
-        assertEquals(exResult + " " + testName.getMethodName() + "-MESSAGE", result);
+        // Using reflection API to set omitEventTypeInLogs field in LogPrefixAppender.
+        setOmitEventTypeInLogsFieldUsingReflection(lpa, omitEventTypeInLogs);
```

Comment:
From my previous comments about making the variable a member instance, we will not need reflection here.
```
System.setProperty(PropNames.OMIT_EVENT_TYPE_IN_LOGS, omitEventTypeInLogs);

 //Since everything is mocked these booleans don't much matter aside from the later verifies
 LogPrefixAppender lpa = new LogPrefixAppender(false, false, false, false, null);
```

Reply:
Except, setting this is a `System` property **_still_** won't make that property get picked up by `ESAPI.securityConfiguration().getBooleanProp()`.

@RodolfoAndre - We have a "trick" to make ESAPI properties work with JUnit tests so we can simulate various values that @jeremiahjstacey may have forgotten about. We use the `org.owasp.esapi.SecurityConfigurationWrapper` class for that. Rather than trying to describe how to use that for JUnit tests, I will instead refer you to
https://github.com/ESAPI/esapi-java-legacy/blob/develop/src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleCleanTest.java
as a typical example of how we use it. If after looking at that, you still can't figure it out, then let me know.

Reply:
No changes needed here as a result of no changes from previous conversations.

Reply:
My initial context is invalid for removing reflection utility, but if we're able to do that I think it would be a good thing.
Thank you for the context @kwwall

Reply:
@kwwall I could change the property just once. After the first initalization the class won't get the property again. So I can't have test for true and false in the same class.

 It will work if I change the test to another file or create a static class inside this one just for testing the flag. Do you have any idea what may I do?

Reply:
@RodolfoAndre - The way I handled it was to create 2 separate JUnit classes, each of which tested one of the possible values. For example, for the property **Validator.HtmlValidationAction**, which can have the value "clean" or "throw", I respectively created:

- https://github.com/ESAPI/esapi-java-legacy/blob/develop/src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleCleanTest.java

- https://github.com/ESAPI/esapi-java-legacy/blob/develop/src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleThrowsTest.java

(although for the latter, it looks as though I may have botched line 53, although I'm not sure it matters since it is set correctly in the `@Before setup` method on line 84).

But that explains the general approach of how I tested both variations. There's probably some other way to do it, but that worked. It's also why the JUnit file are named as they are.  HTH.

Reply:
Ok. So I've added the new file with the tests and reverted the other one just adding a test to check if it is working with the flag off. It is in the new commits that I've sent.